benchmark/internal/arrival/solanaws.go

```diff
@@ -165,22 +186,38 @@ func (s solanaOrderbookStream) Process(updates []arrival.StreamUpdate[solanaRawU
 			return nil
 		})
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 
-		su := solanaUpdate{
-			side:   update.Data.side,
-			orders: orders,
+		side := update.Data.Side
+		su := SolanaUpdate{
+			Side:     side,
+			Orders:   orders,
+			previous: previous[side],
 		}
-
-		results[slot] = append(results[slot], arrival.ProcessedUpdate[solanaUpdate]{
+		pu := ProcessedUpdate[SolanaUpdate]{
 			Timestamp: update.Timestamp,
 			Slot:      slot,
 			Data:      su,
-		})
+		}
+
+		redundant := su.IsRedundant()
+		if redundant {
+			duplicates[slot] = append(results[slot], pu)
```

Comment:
Shouldn't this be `append(duplicates[slot])`...?
```suggestion
			duplicates[slot] = append(duplicates[slot], pu)
```

Reply:
yep, mistake from me